pkg/controller/usbdevice/usbdevice_controller.go

```diff
@@ -0,0 +1,217 @@
+package usbdevice
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/rancher/wrangler/pkg/relatedresource"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/harvester/pcidevices/pkg/apis/devices.harvesterhci.io/v1beta1"
+	"github.com/harvester/pcidevices/pkg/deviceplugins"
+	ctldevicerv1vbeta1 "github.com/harvester/pcidevices/pkg/generated/controllers/devices.harvesterhci.io/v1beta1"
+	"github.com/harvester/pcidevices/pkg/util/gousb"
+	"github.com/harvester/pcidevices/pkg/util/gousb/usbid"
+)
+
+type Handler struct {
+	usbClient      ctldevicerv1vbeta1.USBDeviceClient
+	usbClaimClient ctldevicerv1vbeta1.USBDeviceClaimClient
+}
+
+var walkUSBDevices = deviceplugins.WalkUSBDevices
+
+type USBDevice struct {
+	Name         string
+	Manufacturer string
+	Vendor       int
+	Product      int
+	BCD          int
+	Bus          int
+	DeviceNumber int
+	Serial       string
+	DevicePath   string
+}
+
+func (dev *USBDevice) GetID() string {
+	return fmt.Sprintf("%04x:%04x-%02d:%02d", dev.Vendor, dev.Product, dev.Bus, dev.DeviceNumber)
+}
+
+func NewHandler(usbClient ctldevicerv1vbeta1.USBDeviceClient, usbClaimClient ctldevicerv1vbeta1.USBDeviceClaimClient) *Handler {
+	return &Handler{
+		usbClient:      usbClient,
+		usbClaimClient: usbClaimClient,
+	}
+}
+
+func (h *Handler) OnDeviceChange(_ string, _ string, obj runtime.Object) ([]relatedresource.Key, error) {
+	ud, ok := obj.(*v1beta1.USBDevice)
+
+	if ud == nil {
+		return nil, nil
+	}
+
+	if !ok {
+		logrus.Errorf("error casting object to USBDevice: %v", obj)
+		return nil, nil
+	}
+
+	if ud.Status.NodeName == cl.nodeName {
+		udcList, err := h.usbClaimClient.List(metav1.ListOptions{LabelSelector: cl.selector()})
+		if err != nil {
+			logrus.Errorf("error listing USBDeviceClaims during device watch: %v", err)
+			return nil, err
+		}
+
+		var rr []relatedresource.Key
```

Comment:
we are requeueing all usbdeviceclaims. should we not filter on only the claim where pcidevice address matches the usb device pcidevice address? we could even label the usbdeviceclaim with pcideviceaddress string to make it easy to filter using label selectors.